lib/ansible/modules/network/frr/frr_facts.py

```diff
@@ -0,0 +1,470 @@
+            else:
+                match = re.match(r'^Interface (\S+)', line)
+                if match:
+                    key = match.group(1)
+                    parsed[key] = line
+        return parsed
+
+    def populate_interfaces(self, interfaces):
+        facts = dict()
+        counters = ['description', 'macaddress', 'type', 'vrf', 'mtu', 'bandwidth', 'lineprotocol',
+                    'operstatus']
+        for key, value in iteritems(interfaces):
+            intf = dict()
+            for x in counters:
+                meth = getattr(self, 'parse_%s' % x, None)
+                intf[x] = meth(value)
+            facts[key] = intf
+        return facts
+
+    def parse_description(self, data):
+        match = re.search(r'Description: (.+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_macaddress(self, data):
+        match = re.search(r'HWaddr: (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_type(self, data):
+        match = re.search(r'Type: (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_vrf(self, data):
+        match = re.search(r'vrf: (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_mtu(self, data):
+        match = re.search(r'mtu (\d+)', data)
+        if match:
+            return int(match.group(1))
+
+    def parse_bandwidth(self, data):
+        match = re.search(r'bandwidth (\d+)', data)
+        if match:
+            return int(match.group(1))
+
+    def parse_lineprotocol(self, data):
+        match = re.search(r'line protocol is (\S+)', data)
+        if match:
+            return match.group(1)
+
+    def parse_operstatus(self, data):
+        match = re.search(r'^(?:.+) is (.+),', data, re.M)
+        if match:
+            return match.group(1)
+
+    def populate_ipv4_interfaces(self, data):
+        for key, value in data.items():
+            self.facts['interfaces'][key]['ipv4'] = list()
+            primary_address = addresses = []
+            primary_address = re.findall(r'inet (\S+) broadcast (?:\S+)(?:\s{2,})', value, re.M)
+            addresses = re.findall(r'inet (\S+) broadcast (?:\S+)(?:\s+)secondary', value, re.M)
+            if len(primary_address) == 0:
+                continue
+            addresses.append(primary_address[0])
+            for address in addresses:
+                addr, subnet = address.split("/")
+                ipv4 = dict(address=addr.strip(), subnet=subnet.strip())
+                self.add_ip_address(addr.strip(), 'ipv4')
+                self.facts['interfaces'][key]['ipv4'].append(ipv4)
+
+    def populate_ipv6_interfaces(self, data):
+        for key, value in data.items():
+            self.facts['interfaces'][key]['ipv6'] = list()
+            addresses = re.findall(r'inet6 (\S+)', value, re.M)
+            for address in addresses:
+                addr, subnet = address.split("/")
```

Comment:
same as above

Reply:
@ganeshrn Could not find a matching method that could be used in this scenario.